tools/assets_preprocessor/src/assets_preprocessor.h

```diff
@@ -48,10 +36,239 @@
+ * to configure how a single asset file or a directory of asset files is processed.
+ * Different asset processor functions may have different options or no options at all,
+ * and the specific options supported by each one are listed in the next section.
+ *
+ * To apply options to a specific file, e.g. `myImage.png`, create another file in the
+ * same directory, but with the extension `.opts` instead, e.g. `myImage.opts`. To apply
+ * options to an entire directory and all its subdirectories, create a file called `.opts`
+ * inside that directory.
+ *
+ * Only a single options file will be used when processing any particular file; options
+ * files are never merged. When an input file is being processed, the assets preprocessor
+ * first searches for the `<filename>.opts` file, and loads its options if it exists. If
+ * that options file does not exist, then the preprocessor will search for the `.opts` file
+ * in the same directory as the input file. If that does not exist, it will search the
+ * directory containing the input file, and so on until the top-level input directory has
+ * been searched. The first of these `.opts` file that exists will be used and no other
+ * assets files will be searched.
+ *
+ * An options file is an INI-style file similar to the config file, but instead of each
+ * section defining a file extension mapping, in an options file each section contains
+ * a list of options for a specific asset processor. The section name, e.g. `[wsg]` or
+ * `[heatshrink]` should match the name of the asset processor function being used.
+ * Multiple sections may be included in a single options file, though this only makes
+ * sense for `.opts` files in a directory containing multiple input file types.
+ *
+ * \subsubsection assetProc_optionsExample Options File Example
+ *
+ * ```ini
+ * ; In an Options file, the section name should match the function name
+ * [wsg]
+ * dither = true
+ * ```
+ *
+ * \subsubsection assetProc_funcs Function Asset Preprocessors
+ *
+ * For all available asset processing functions, run the asset processor with the
+ * `-h` option. Here is a list of the currently available processors and a brief
+ * description of them, along with any options they support.
+ *
+ * \paragraph assetProc_bin bin
+ * Copies the input file directly to the output file with no changes.
+ *
+ * \paragraph assetProc_chart chart
+ * Processes the input file as a Clone Hero chart, which can be created by
+ * <a href="https://efhiii.github.io/midi-ch/">this tool</a>. See also the
+ * <a
+ * href="https://github.com/TheNathannator/GuitarGame_ChartFormats/blob/main/doc/FileFormats/.chart/Core%20Infrastructure.md">
+ * .chart file spec</a>.
+ *
+ * \paragraph assetProc_font font
+ * Processes special font PNG files, which can be created by the
+ * <a href="https://github.com/AEFeinstein/Super-2024-Swadge-FW/blob/main/tools/font_maker/README.md">
```

Comment:
We can include this `README.md` in doxygen, but if only we could link it sanely :(

Reply:
Leaving this here, there's a few options I don't think I quite tried before so maybe we can fix this after all...

https://stackoverflow.com/a/26710792

Reply:
You can try, but it feels the issue here is specific to it being a `README.md` not `RANDOM_NAME.md` and Doxygen treating it like directory do.

Wait, maybe this? https://stackoverflow.com/a/37716243